Feat: 1.3 new create import srp

ieow opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

1.3
Update New SRP onboarding flow as per figma design
https://www.figma.com/design/pViOUcmjwhEzFsdrwknpNc/Onboarding-redesign?node-id=434-55200&t=wmEoQSF9g9lFtkwp-0

This PR is part 3 of 7 for the new SRP ui update

This PR cover

 - new ui create srp
 - new ui for import srp

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

https://github.com/user-attachments/assets/a4e8578a-63a1-4393-8237-ef2b339a34bd


## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

ieow committed:
feat: new onboarding carousel and use terms

Co-authored-by: smgv <gpatra1996@gmail.com>

ieow committed:
feat: new welcome screen - let get started

ieow committed:
feat: new onboarding carousel and use terms

Co-authored-by: smgv <gpatra1996@gmail.com>

ieow committed:
feat: new UI for choose password and importFromSeed

Co-authored-by: smgv <gpatra1996@gmail.com>

ieow committed:
fix: import srp seed grid and focus

ieow committed:
fix: address comment

ieow committed:
fix: remove duplicated code

smgv committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

ieow committed:
fix: comment

smgv committed:
Fix: ModalMandatory Test Case

ieow committed:
fix: address comment
update to use 8 for macos, and rename variables

chaitanyapotti committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

smgv committed:
Fix: Onboarding Carousel Test case Update

smgv committed:
Merge branch 'feat/srp-onboarding-carousel-tnc' of https://github.com/MetaMask/metamask-mobile into feat/srp-onboarding-carousel-tnc

chaitanyapotti committed:
Merge branch 'feat/srp-onboarding-carousel-tnc' into feat/new-srp-welcome

chaitanyapotti commented on app/components/Views/ImportFromSecretRecoveryPhrase/styles.ts:
```diff
@@ -1,92 +1,118 @@
-    ctaWrapper: {
-      marginTop: 20,
+    seedPhraseInnerContainer: {
+      paddingHorizontal: Platform.OS === 'ios' ? 16 : 14,
```
use Platform.select

ieow committed:
fix: address comment

chaitanyapotti commented on app/components/Views/ChoosePassword/index.js:
```diff
@@ -242,11 +212,13 @@ class ChoosePassword extends PureComponent {
     secureTextEntry: true,
     biometryType: null,
-    biometryChoice: false,
+    biometryChoice: true,
```
no need to set this to true. the existing logic determines based on checks

chaitanyapotti commented on app/components/Views/ChoosePassword/index.js:
```diff
@@ -548,8 +539,22 @@ class ChoosePassword extends PureComponent {
+
   setConfirmPassword = (val) => this.setState({ confirmPassword: val });
 
+  isError = () => {
```
rename this

chaitanyapotti commented on app/components/Views/ChoosePassword/index.js:
```diff
@@ -597,166 +609,183 @@ class ChoosePassword extends PureComponent {
+                      secureTextEntry={this.state.showPasswordIndex.includes(0)}
+                      placeholderTextColor={colors.text.muted}
+                      testID={ChoosePasswordSelectorsIDs.NEW_PASSWORD_INPUT_ID}
+                      onSubmitEditing={this.jumpToConfirmPassword}
```
use autoComplete tags

chaitanyapotti replied:
```suggestion
                      onSubmitEditing={this.jumpToConfirmPassword}
                      autoComplete="new-password"
```

chaitanyapotti commented on app/components/Views/ImportFromSecretRecoveryPhrase/index.js:
```diff
@@ -86,37 +111,191 @@ const ImportFromSecretRecoveryPhrase = ({
+
+  const seedPhraseLength = seedPhrase.filter((item) => item !== '').length;
+
+  const isSRPContinueButtonDisabled = () =>
```
copy the logic on extension. it's better

chaitanyapotti replied:
Will do this in a later PR

smgv committed:
Fix: Onboarding Carousel Change Tab Test Case Added

ieow committed:
fix: save tracking event prior optin screen
fix: getTransparentOnboardingNavbarOptions

ieow committed:
fix: revert deleted css

smgv committed:
Fix: Onboarding Carousel Update ChangeTab

smgv committed:
Fix: Navbar getOnboardingCarouselNavbarOptions Test Case

ieow committed:
fix: address comments

ieow committed:
Merge branch 'feat/new-srp-welcome' into feat/new-create-import-srp

chaitanyapotti committed:
Merge branch 'feat/srp-onboarding-carousel-tnc' into feat/new-srp-welcome

smgv committed:
Fix: Test Case for OnBoardingScreenWithBg and TermsOfUse added

smgv committed:
Fix: ModalMandatory Test Case Updated

ieow committed:
fix: comment issue

ieow committed:
fix: locale

chaitanyapotti committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

smgv committed:
Fix: Test case update for PR 1.1

smgv committed:
Merge branch 'feat/srp-onboarding-carousel-tnc' of https://github.com/MetaMask/metamask-mobile into feat/srp-onboarding-carousel-tnc

smgv committed:
Fix: Code Fix nested ternary operation

ieow committed:
fix: update snapshot

chaitanyapotti committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

chaitanyapotti committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

hieu-w committed:
test: fix e2e acceptTermOfUse right after tap Started

hieu-w committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

hieu-w committed:
test: use launchApp instead openURL w only support iOS

smgv committed:
Fix: Choose Password Test Case Update

smgv committed:
Fix: Onboarding Test Case Update

smgv committed:
Fix: Navbar Test Case Updated

smgv committed:
Fix: Navbar Test Case Update For ImportFromSecretRecoveryPhrase

hieu-w committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

hieu-w committed:
test: move acceptTermOfUse to right point

hieu-w committed:
test: skip wrong e2e implement

hieu-w committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

hieu-w committed:
Revert "test: skip wrong e2e implement"

This reverts commit 88015e32bc491805ee6cdc4bb39b5c9ebe304e94.

smgv committed:
Fix: ImportFromSecretRecoveryPhrase from current step 2 Test Case Updated

ieow committed:
Merge branch 'feat/new-srp-welcome' into feat/new-create-import-srp

hieu-w committed:
Merge branch 'feat/srp-onboarding-carousel-tnc' into feat/new-srp-welcome

ieow committed:
fix: update dispatch save onboarding event

hieu-w committed:
test: e2e quarantine "send eth test"

hieu-w committed:
Merge branch 'feat/srp-onboarding-carousel-tnc' into feat/new-srp-welcome

hieu-w committed:
test: e2e quarantine send-eth test

hieu-w committed:
Merge branch 'feat/srp-onboarding-carousel-tnc' into feat/new-srp-welcome

smgv committed:
Fix: Test Case Update For BackButton and QR code removed unused code

smgv committed:
Fix: test Case update for events

smgv committed:
Fix: Test Case Update

hieu-w committed:
test: update Matcher for importSeedButton

hieu-w committed:
test: handle case when MetricsOptin not shown

smgv committed:
Fix: Import Wallet Test Case Update

smgv committed:
Fix: removed not required condition

hieu-w committed:
test: skip sync account srp test until implemented

chaitanyapotti committed:
Merge branch 'feat/new-srp-welcome' into feat/new-create-import-srp

ieow commented on app/components/Views/ImportFromSecretRecoveryPhrase/index.js:
```diff
@@ -245,10 +531,32 @@ const ImportFromSecretRecoveryPhrase = ({
-          routes: [{ name: Routes.ONBOARDING.SUCCESS_FLOW }],
-        });
+
+        if (isMetricsEnabled()) {
```
For reseted wallet, the metrics' enable  is still kept 
like the terms of use cases.
This is the previous metric behaviour

ieow commented on app/components/Views/ImportFromSecretRecoveryPhrase/index.js:
```diff
@@ -83,40 +106,199 @@ const ImportFromSecretRecoveryPhrase = ({
     const eventBuilder = MetricsEventBuilder.createEventBuilder(event);
     eventBuilder.addProperties(properties);
-    trackOnboarding(eventBuilder.build());
+    trackOnboarding(eventBuilder.build(), dispatchSaveOnboardingEvent);
```
save events to redux prior consents.
will discard or sync to metric once user navigate to optin metric screen

smgv committed:
Fix: TestCase Update Choose Password

smgv committed:
Merge branch 'feat/new-create-import-srp' of https://github.com/MetaMask/metamask-mobile into feat/new-create-import-srp

smgv committed:
Fix: HandleKeyPress Test Case Update

hieu-w committed:
test: e2e update importWalletWithRecoveryPhrase steps

smgv committed:
Fix: validateSeedPhrase  test case update

smgv committed:
Merge branch 'feat/new-create-import-srp' of https://github.com/MetaMask/metamask-mobile into feat/new-create-import-srp

smgv committed:
Fix: Test Case Fail Update

smgv committed:
Fix: toggleShowPassword code maintainability

smgv committed:
Fix: handleFocus test case update

hieu-w committed:
test: fix testId in onboarding flow srp

hieu-w committed:
Merge branch 'feat/new-create-import-srp' of https://github.com/MetaMask/metamask-mobile into feat/new-create-import-srp

ieow committed:
feat: revert optin metric navigation.
will add another pr at the end

hieu-w committed:
feat: revert SRP util to previous, will modify flow later

hieu-w committed:
test: revert skip test in account-syncing

smgv committed:
Fix: Comment Review

hieu-w committed:
Merge branch 'feat/new-srp-welcome' into feat/new-create-import-srp

ieow committed:
fix: revert optin navigation
fix delayed optin in another pr

hieu-w committed:
Merge branch 'feat/new-create-import-srp' of https://github.com/MetaMask/metamask-mobile into feat/new-create-import-srp

hieu-w committed:
test: remove duplicate Optin login in e2e srp function

ieow committed:
fix: lint , lint:tsc

hieu-w committed:
test: add delay after enter SRP

hieu-w committed:
Merge branch 'feat/new-create-import-srp' of https://github.com/MetaMask/metamask-mobile into feat/new-create-import-srp

hieu-w committed:
Merge branch 'main' into feat/new-create-import-srp

hieu-w committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

hieu-w committed:
Merge branch 'feat/srp-onboarding-carousel-tnc' into feat/new-srp-welcome

hieu-w committed:
Merge branch 'feat/new-srp-welcome' into feat/new-create-import-srp

hieu-w committed:
test: fix duplicate testId cause Issue on ios srp flow

ieow committed:
feat: New welcome screen (#15666)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**
Update New SRP onboarding flow as per figma design

https://www.figma.com/design/pViOUcmjwhEzFsdrwknpNc/Onboarding-redesign?node-id=434-55200&t=wmEoQSF9g9lFtkwp-0

This PR is part 2 of 7 for the new SRP ui update

This PR covers

- new ui for welcome page

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Related issues**

Fixes:

## **Manual testing steps**

1. After opening app, agree to terms
2. Come to the Let's get started page

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

https://github.com/user-attachments/assets/2e37d1a2-3587-42d5-b90a-9965b5168ed2


## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: Chaitanya Potti <chaitanya.potti@gmail.com>
Co-authored-by: smgv <gpatra1996@gmail.com>
Co-authored-by: hieu-w <maestrokoder@gmail.com>
Co-authored-by: hieu-w <hieu@tor.us>

hieu-w committed:
test: fix ios typing fast cause issue on srp flow

hieu-w committed:
Merge branch 'feat/srp-onboarding-carousel-tnc' into feat/new-create-import-srp

ieow committed:
fix: add delay to wait grid srp render and button enabled

ieow committed:
fix: add logging for debug

ieow committed:
fix: add more delay before tap continue

chaitanyapotti committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

ieow committed:
fix: waitFor if enable

hieu-w committed:
test: fix keyboard overlay issue on SRP screen

ieow committed:
fix: add more threshold padding allowance for android in optin matric

ieow committed:
Merge branch 'feat/srp-onboarding-carousel-tnc' into feat/new-create-import-srp

ieow committed:
feat: optin button sometime will always disabled even swipe to end
remove disable for now

ieow committed:
fix: update snapshot

ieow committed:
fix: update optin metric ui

ieow committed:
fix: update snapshot

chaitanyapotti committed:
performance optimizations

chaitanyapotti reviewed (dismissed):
lgtm

ieow committed:
fix: tests - error message logic update

ieow committed:
chore: add test case on handle new input

smgv committed:
Fix: Test Case Update and Lint issue Update

smgv committed:
Fix: removed unused variable

brianacnguyen commented on app/component-library/components/Icons/Icon/Icon.types.ts:
```diff
@@ -343,5 +343,7 @@ export enum IconName {
   WifiOff = 'WifiOff',
   Wifi = 'Wifi',
+  DangerSolid = 'DangerSolid',
+  SuccessSolid = 'SuccessSolid',
```
These do not belong to the Icon component. Please import the images and use them directly

tommasini replied:
Thanks Brian for the review! Just to add an extra info. You can use them directly because we use the react-native-svg-transformer at the metro file: https://github.com/MetaMask/metamask-mobile/blob/8ab142602aae723ce2cd33826de565daee5fabfe/metro.transform.js#L11

tommasini replied:
Move it to assets please, only monochromatic icons should be at this assets folder of component-library, Brian mentioned that in the future we may have one Icon that supports polychromatic svgs

brianacnguyen reviewed (changes requested):
Please remove semantic icons from PR

ieow committed:
Merge branch 'main' into feat/new-create-import-srp

hieu-w committed:
Merge branch 'main' into feat/new-create-import-srp

hieu-w committed:
feat: update test snapshot

tommasini commented on app/components/UI/OptinMetrics/index.js:
```diff
@@ -2,17 +2,16 @@ import React, { PureComponent } from 'react';
 import {
```
This is a class javascript component, since we are changing it, would be a great occasion to refactor it to a typescript functional component!
Since we are trying to hit a deadline, we can address this in a separate PR

tommasini commented on app/components/Views/ChoosePassword/index.test.tsx:
```diff
@@ -1,11 +1,45 @@
+  },
+}));
+
+// Mock LottieView
```
self explanatory code can be removed

smgv replied:
Please review @tommasini

cc: @ieow

tommasini commented on app/components/Views/ChoosePassword/index.test.tsx:
```diff
@@ -1,11 +1,45 @@
+import { ThemeContext, mockTheme } from '../../../util/theme';
+import { ChoosePasswordSelectorsIDs } from '../../../../e2e/selectors/Onboarding/ChoosePassword.selectors';
+
+// Mock Engine
```
self explanatory code, comment can be removed

tommasini commented on app/components/Views/ChoosePassword/index.test.tsx:
```diff
@@ -19,16 +53,254 @@ const initialState = {
+      await new Promise((resolve) => setTimeout(resolve, 100));
+    });
+
+    // // Check if title text is rendered correctly for loading state
```
This test isn't testing anything, were this forgotten?

smgv replied:
Updated. Please review @tommasini

cc: @ieow

tommasini commented on app/components/Views/ChoosePassword/index.test.tsx:
```diff
@@ -19,16 +53,254 @@ const initialState = {
+    // const titleText = component.getByText(
+    //   strings('secure_your_wallet.creating_password'),
+    // );
+    // expect(titleText).toBeTruthy();
```
this test is not testing anything, forgot to uncomment?

smgv replied:
Updated. Please review @tommasini

cc: @ieow

tommasini commented on app/components/Views/ChoosePassword/index.test.tsx:
```diff
@@ -19,16 +53,254 @@ const initialState = {
+      headerLeftComponent.props.onPress();
+    });
+
+    // Verify that goBack was called
```
We can remove comments in self explanatory code

tommasini commented on app/components/Views/ImportFromSecretRecoveryPhrase/index.js:
```diff
@@ -64,13 +63,42 @@ import { ImportFromSeedSelectorsIDs } from '../../../../e2e/selectors/Onboarding
+import { LoginOptionsSwitch } from '../../UI/LoginOptionsSwitch';
+import { saveOnboardingEvent } from '../../../actions/onboarding';
 
 const MINIMUM_SUPPORTED_CLIPBOARD_VERSION = 9;
 
+const SRP_LENGTHS = [12, 15, 18, 21, 24];
+
 const PASSCODE_NOT_SET_ERROR = 'Error: Passcode not set.';
 const IOS_REJECTED_BIOMETRICS_ERROR =
   'Error: The user name or passphrase you entered is not correct.';
 
+const SPACE_CHAR = ' ';
```
All this are constants, should they be moved to a constants file inside the compoennt folder?

smgv replied:
Updated. Please review @tommasini

cc: @ieow

tommasini commented on app/components/Views/ImportFromSecretRecoveryPhrase/index.js:
```diff
@@ -270,342 +557,427 @@ const ImportFromSecretRecoveryPhrase = ({
+                                style={[
+                                  {
+                                    width: containerWidth / 3,
+                                    padding: inputPadding,
```
InputPadding variable is only used here, should we just use the code?
```suggestion
                                    padding: Platform.OS === 'ios' ? 4 : 3,
```

tommasini commented on app/components/Views/ImportFromSecretRecoveryPhrase/index.js:
```diff
@@ -83,40 +111,196 @@ const ImportFromSecretRecoveryPhrase = ({
   seedphraseBackedUp,
   setOnboardingWizardStep,
   route,
+  dispatchSaveOnboardingEvent,
```
Would be so awesome, converting this component to typescript, since this got a major refacator!

smgv committed:
Fix: Comment Review

smgv committed:
Fix: unused code removed

tommasini committed:
Merge branch 'main' into feat/new-create-import-srp

smgv commented on app/components/Views/ChoosePassword/index.test.tsx:
```diff
@@ -1,11 +1,45 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import { render, act, fireEvent } from '@testing-library/react-native';
+// import { ActivityIndicator } from 'react-native';
```
Please review @tommasini 

cc: @ieow

smgv commented on app/components/Views/ChoosePassword/index.test.tsx:
```diff
@@ -1,11 +1,45 @@
+// Mock LottieView
+jest.mock('lottie-react-native', () => 'LottieView');
+
+// Mock Authentication
```
Updated. Please review @tommasini

cc: @ieow

smgv commented on app/components/Views/ImportFromSecretRecoveryPhrase/index.js:
```diff
@@ -83,40 +111,196 @@ const ImportFromSecretRecoveryPhrase = ({
+  const [containerWidth, setContainerWidth] = useState(0);
+
+  const inputPadding = Platform.OS === 'ios' ? 4 : 3;
+  const numColumns = 3; // Number of columns
```
Updated. Please review @tommasini

cc: @ieow

smgv commented on app/components/Views/ImportFromSecretRecoveryPhrase/index.js:
```diff
@@ -270,342 +557,427 @@ const ImportFromSecretRecoveryPhrase = ({
+                                    handleSeedPhraseChange(text, index)
+                                  }
+                                  placeholderTextColor={colors.text.muted}
+                                  // autoFocus={index === seedPhrase.length - 1}
```
I just wanted to let you know that this wasn't required.  @tommasini

cc: @ieow

smgv committed:
Fix: Comment Review Icon

smgv committed:
Merge branch 'feat/new-create-import-srp' of https://github.com/MetaMask/metamask-mobile into feat/new-create-import-srp

smgv committed:
Fix: removed unused imports

hieu-w committed:
Merge branch 'main' into feat/new-create-import-srp

smgv committed:
Fix: Moved Icons from asset to images folder
